### PR TITLE
go.mod: update dependencies

### PR DESCRIPTION
**Pull request by avalluri**
result of `go get -u` and `go get k8s.io/client-go@v0.24.0-beta.0`

### CODE REVIEW

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (ipuustin) [inline api/v1alpha1/zz_generated.deepcopy.go:1]: Was this also done by the go dependency updates? Or is adding this change accidental?

Comment (avalluri) [reply api/v1alpha1/zz_generated.deepcopy.go:1]: I am not sure. When I build after updating dependencies it ended up with this diff.

Comment (ipuustin) [reply api/v1alpha1/zz_generated.deepcopy.go:1]: It seems that both tags need to be present (the first one for gofmt >= 1.17 and the second one for compatibility), so I think the change is anyway good: https://github.com/kubernetes/gengo/pull/210

**Review by ipuustin — APPROVED** [review]
LGTM.

**Pull request merged**